src/app/core/public/api/impl/Circuit.ts

```diff
@@ -118,14 +115,18 @@ export abstract class CircuitImpl<
     }
 
     // Queries
-    public pickObjectAt(pt: Vector): ComponentT | WireT | PortT | undefined {
-        throw new Error("Unimplemented");
+    public pickObjectAt(pt: Vector, space: Vector.Spaces = "world"): ComponentT | WireT | PortT | undefined {
+        const pos = ((space === "world") ? pt : this.view.toWorldPos(pt));
+        const objID = this.view.findNearestObj(pos);
+        if (!objID)
+            return undefined;
+        return this.getObj(objID);
```

Comment:
Could be
```typescript
return !objID ? undefined : this.getObj(objID);
```
Can't remember whether you liked ternary for things like this

Reply:
eh yeah not really in this case, I think undefined checks should remain if-statements for the most part